front/lib/api/assistant/conversation.ts

```diff
@@ -1427,6 +1461,66 @@ export async function* retryAgentMessage(
   yield* streamRunAgentEvents(auth, eventStream, agentMessage, agentMessageRow);
 }
 
+// Injects a new content fragment in the conversation.
+export async function postNewContentFragment(
+  auth: Authenticator,
+  conversation: ConversationType,
+  content: string
+): Promise<Result<ContentFragmentType, Error>> {
```

Comment:
We don't need a result IMHO here as you are catching to emit an unqualified 500 but really we can just not catch and let the middleware return the same 500?

This will remove a few lines of code?

Reply:
That's what I did initially,  but felt wrong to try/catch in the endpoint since we said we never want to do that ?